crates/circuit/src/circuit_data.rs

```diff
@@ -2372,73 +2318,11 @@ impl CircuitData {
-                            // argument position 1.
-                            if let Some(loop_param) = loop_param {
-                                self.param_table.track(
-                                    &loop_param.bind(py).extract()?,
-                                    Some(ParameterUse::Index {
-                                        instruction: instruction_index,
-                                        parameter: 1,
-                                    }),
-                                )?;
-                            }
-                            let usage = ParameterUse::Index {
-                                instruction: instruction_index,
-                                parameter: 2,
-                            };
-                            for param_ob in &blocks[0]
-                                .bind(py)
-                                .getattr(intern!(py, "parameters"))?
-                                .try_iter()?
-                            {
-                                self.param_table.track(&param_ob?.extract()?, Some(usage))?;
-                            }
-                            Ok(())
-                        })?;
-                    }
-                    OperationRef::ControlFlow(_) => {
-                        // All the other control flow operations are simple, and their "blocks"
-                        // are exactly their parameters, in the right order.
-                        let blocks: Vec<_> = self
-                            .try_view_control_flow(instr)
-                            .unwrap()
-                            .blocks()
-                            .into_iter()
-                            .cloned()
-                            .collect();
-                        Python::attach(|py| -> PyResult<_> {
-                            for (idx, case) in blocks.iter().enumerate() {
-                                let usage = ParameterUse::Index {
-                                    instruction: instruction_index,
-                                    parameter: idx as u32,
-                                };
-                                for param_ob in case
-                                    .bind(py)
-                                    .getattr(intern!(py, "parameters"))?
-                                    .try_iter()?
-                                {
-                                    self.param_table.track(&param_ob?.extract()?, Some(usage))?;
-                                }
-                            }
-                            Ok(())
-                        })?;
-                    }
-                    _ => panic!("instruction has blocks but is not control flow"),
-                }
+                let cf = ControlFlowView::<CircuitData>::try_view(instr, &self.blocks)
+                    .expect("instructions that have blocks are control flow");
```

Comment:
Should the message here be a little more descriptive? AFAIA the `try_view()` only returns `None` if the operation is anything but a Control Flow Op. Perhaps we can do away with:

```suggestion
                    .expect("An invalid non-control flow operation with blocks was provided.");
```
Though it may still not be explicit enough. The same goes on the next block with the same message.

Reply:
An `expect` or `unwrap` is an abort condition, which typically should only happen if some program invariant has been violated.  If we're in that situation, then _presumably_ there's an internal logic error, so we can't necessarily know that the data structures are internally coherent, which means we might not be correct if we try and say _what_ has gone wrong.

With that in mind, [idiomatic Rust suggests that the `expect` message explains why the _programmer_ expected it to work](https://doc.rust-lang.org/std/option/enum.Option.html#recommended-message-style).  So the reason I expect that this `expect` call works is that the only instructions that have blocks associated with them _are_ control-flow blocks, so now if this line triggers, the person debugging it will understand why I thought it was ok to write `expect` here, and they can see if the problem is my assumption was wrong, my assumption is out-dated, or there's some other internal-logic error and the `expect` _is_ correct.